components/sup/build.rs

```diff
@@ -33,3 +27,28 @@ fn generate_apidocs() {
         }
     };
 }
+
+#[cfg(windows)]
+// One would think we could directly call the .bat file
+// see https://github.com/rust-lang/rust/issues/42791
+// for why this does not work
+fn raml2html_cmd(dst: PathBuf, src: PathBuf) -> io::Result<ExitStatus> {
+    Command::new("cmd")
+        .arg("/c")
+        .arg("raml2html.bat")
+        .arg("-i")
+        .arg(src)
+        .arg("-o")
+        .arg(dst)
+        .status()
+}
+
+#[cfg(unix)]
+fn raml2html_cmd() -> io::Result<ExitStatus> {
```

Comment:
Where are `src` and `dst` coming from?

In the spirit of making it clear what parts are different per-platform and catching bugs at compile time, I think we can have one implementation:
```rust
fn raml2html_cmd(dst: PathBuf, src: PathBuf) -> io::Result<ExitStatus> {
    let mut cmd = Command::new(if cfg!(windows) { "cmd" } else { "raml2html" });

    let cmd = if cfg!(windows) {
        cmd.arg("/c").arg("raml2html.bat")
    } else {
        &mut cmd
    };

    cmd.arg("-i").arg(src).arg("-o").arg(dst).status()
}
```

The need for two separate `if cfg!` blocks owes to the fact that the builder pattern returns different types for `new` and `arg`. I think this is the cleanest way to do it.

Reply:
Seems like a good simplification 👍